script/prerender/fileUtils.ts

```diff
@@ -42,3 +47,46 @@ export const createDiskCache = <K extends string, V>(prefix: string): Cache<K, V
     },
   };
 };
+
+let s3Client: S3Client | undefined;
+
+// Generates a release path for a file without a leading /, used when uploading the release to S3
+function prefixReleasePath(filepath: string) {
+  let basePath = assertDefined(process.env.PUBLIC_URL, 'PUBLIC_URL environment variable not set');
+  if (basePath[0] === '/') { basePath = basePath.slice(1); }
+  return `${basePath}${filepath}`;
+}
+
+async function writeS3File(key: string, contents: string, contentType: string) {
+  if (!s3Client) {
+    // We explicitly pass credentials to the S3 Client only once
+    // so it'll not try to load them over and over
+
+    console.log('Fetching container credentials');
+    const credentials = await fromContainerMetadata();
+
+    console.log('Initializing S3 client');
+    s3Client = new S3Client({ credentials, region: process.env.BUCKET_REGION });
+  }
```

Comment:
there is some danger of doing this twice concurrently, you could also do it like
```
const getS3Client = once(async () => {
    console.log('Fetching container credentials');
    const credentials = await fromContainerMetadata();

    console.log('Initializing S3 client');
    return new S3Client({ credentials, region: process.env.BUCKET_REGION });
})

async function writeS3File(key: string, contents: string, contentType: string) {
  const s3Client = await getS3Client();
}


```

this way even if its called concurrently both usages will get the same promise

Reply:
Ah interesting... ok I'll add this